x-pack/legacy/plugins/ml/public/data_frame/pages/data_frame_new_pivot/components/step_define/step_define_form.tsx

```diff
@@ -441,19 +451,17 @@ export const StepDefineForm: SFC<Props> = React.memo(({ overrides = {}, onChange
                     defaultMessage: 'Use a query string to filter the source data (optional).',
                   })}
                 >
-                  <EuiSearchBar
-                    defaultQuery={search === defaultSearch ? emptySearch : search}
-                    box={{
-                      placeholder: i18n.translate(
-                        'xpack.ml.dataframe.stepDefineForm.queryPlaceholder',
-                        {
-                          defaultMessage: 'e.g. {example}',
-                          values: { example: 'method:GET -is:active' },
-                        }
-                      ),
-                      incremental: false,
-                    }}
-                    onChange={searchHandler}
+                  <KqlFilterBar
+                    indexPattern={indexPattern}
+                    onSubmit={searchHandler}
+                    initialValue={search === defaultSearch ? emptySearch : search}
+                    placeholder={i18n.translate(
+                      'xpack.ml.dataframe.stepDefineForm.queryPlaceholder',
+                      {
+                        defaultMessage: 'e.g. {example}',
+                        values: { example: 'method:GET -is:active' },
```

Comment:
The `-is:active` part of the placeholder is not valid KQL is it? I'd remove that part of the placeholder and either just leave `method:GET` or make it something like `method : "GET" or status : "404"`

Reply:
Good call - updated in 0f1f18012dc0bd4505fa88b6d113eb080915314a